merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostgresParsers.java

```diff
@@ -95,9 +96,21 @@ public JsonValue unparse(final Timestamp value) {
           realProfileTypeP);
 
   public static Duration parseOffset(final ResultSet resultSet, final int index, final Timestamp epoch) throws SQLException {
-    final var interval = Interval.parse(resultSet.getString(index));
-    final Timestamp end = new Timestamp((Instant)interval.addTo(epoch.toInstant()));
-    return Duration.of(epoch.microsUntil(end), Duration.MICROSECONDS);
+    return Duration.of(microsOfPGInterval(new PGInterval(resultSet.getString(index))), Duration.MICROSECONDS);
+  }
+
+  static long microsOfPGInterval(final PGInterval interval) {
+    if (interval.getYears() != 0) {
+      throw new IllegalArgumentException("Cannot convert years to microseconds");
+    }
+    if (interval.getMonths() != 0) {
+      throw new IllegalArgumentException("Cannot convert months to microseconds");
+    }
+    return interval.getMicroSeconds() +
+           (1_000_000L * interval.getWholeSeconds()) +
+           (1_000_000L * 60 * interval.getMinutes()) +
+           (1_000_000L * 3600 * interval.getHours()) +
+           (1_000_000L * 3600 * 24 * interval.getDays());
```

Comment:
How does this work if this is called to parse, for example, the start offset of a span that starts two months into a year long simulation dataset? It looks like the method would throw an `IllegalArgumentException`, which I'd argue is unexpected behavior for calling [`GetSpanRecords.get()`](https://github.com/NASA-AMMOS/aerie/blob/prototype/use-org-postgresql-jdbc/merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/GetSpanRecords.java#L50) on a valid dataset.

Reply:
Awesome! Would it be possible to include this check as a test for the `microsOfPGInterval` function?

I agree that further investigation is a separate body of work, and just including that test would be sufficient to resolve the concern my initial comment brought up.

Reply:
Could you help me understand which check you're referring to?

Reply:
The check where span records with a start offset greater than or equal to a month/year are produced and then fetched via `GetSpanRecords.get()`.